test/type/opaque.swift

```diff
@@ -543,5 +547,23 @@ func sr10988() {
       x // expected-note {{did you mean to return the last expression?}} {{7-7=return }}
       // expected-warning@-1 {{expression of type 'Int' is unused}}
     }
+    func test2() -> some Numeric {
+      // expected-error@-1 {{function declares an opaque return type, but has no return statements in its body from which to infer an underlying type}}
+      let x = "s" // expected-warning {{initialization of immutable value 'x' was never used; consider replacing with assignment to '_' or removing it}}
+    }
+  }
+  struct S2: P1, P2 {
+  }
+  func test3() -> some P1 & P2 {
+    // expected-error@-1 {{function declares an opaque return type, but has no return statements in its body from which to infer an underlying type}}
+    let x = S2()
+    x // expected-note {{did you mean to return the last expression?}} {{5-5=return }}
+    // expected-warning@-1 {{expression of type 'S2' is unused}}
+  }
+  struct S3: P1 {
+    func test3() -> some P1 & P2 {
+      // expected-error@-1 {{function declares an opaque return type, but has no return statements in its body from which to infer an underlying type}}
+      let x = S2() // expected-warning {{initialization of immutable value 'x' was never used; consider replacing with assignment to '_' or removing it}}
```

Comment:
I think you'd want to have `x` as a last expression here as well so the logic you added is exercised.

Reply:
Right, will just fix it too.